### PR TITLE
v1.2.x

### PR DESCRIPTION
**Pull request by StoneT2000**

### CODE REVIEW

**Comment by StoneT2000** [comment]
most recent pr addresses #58

**Review by StoneT2000 — COMMENTED** [review]
just reviewed all files and ran tests, up until this point changes are fine.

File: src/Game/index.ts [hunk 1]
```diff
@@ -488,10 +460,16 @@ export class Game {
 
   spawnWorker(team: Unit.TEAM, x: number, y: number, unitid?: string): Worker {
     const cell = this.map.getCell(x, y);
-    const unit = new Worker(x, y, team, this.configs, this.globalUnitIDCount++);
+    const unit = new Worker(
+      x,
+      y,
+      team,
+      this.configs,
+      this.globalUnitIDCount + 1
```
Comment (lgloria) [inline src/Game/index.ts:468]: @StoneT2000 Is this intentional? Look likes an off by 1 error.

Comment (StoneT2000) [reply src/Game/index.ts:468]: this is intentional. We increment this count later if we aren't given a id to use. If we are given an id to use, this is purely for resetting state and we expect the global is to be set elsewhere when resetting set (see toStateObj and the information stored there)

**Pull request merged**